app/controllers/callbacks/steps_controller.rb

```diff
@@ -50,7 +51,9 @@ def wizard_store
     end
 
     def app_store
-      session[:callbacks] ||= {}
+      redirect_to session_expired_path if session[:mailinglist].blank?
```

Comment:
Not sure if it makes sense to redirect in the `app_store` method; would it be cleaner to have a `before_action` that checks if the session has expired?

Reply:
good point, thanks